airflow/dag_processing/manager.py

```diff
@@ -526,6 +526,12 @@ def deactivate_stale_dags(
         dags_parsed = session.execute(query)
 
         for dag in dags_parsed:
+            # When the DAG processor runs as part of the scheduler, and the user changes the DAG home folder,
+            # DAGs from the previous DAG home will be marked as stale. Note that this change has no impact
+            # on standalone DAG processors.
+            if os.path.commonpath([dag.fileloc, dag_directory]) != dag_directory:
```

Comment:
I almost feel this should just be an additional condition to the next if block - that way we still log that the DAG is missing for this case too.

Reply:
I think it will be a little difficult to read, because it would look like - 
```
if (
                os.path.commonpath([dag.fileloc, dag_directory]) != dag_directory
                or
                (dag.fileloc in last_parsed
                and (dag.last_parsed_time + timedelta(seconds=stale_dag_threshold)) < last_parsed[dag.fileloc])
):
```
We can probably add `cls.logger().info("DAG %s is missing and will be deactivated.", dag.dag_id)` to previous if condition?

Reply:
Right, but you can mitigate it by doing something like this (no time spent thinking of naming):

```
            # explain different DAGs folder in a comment
            dag_folder_different = ...
            # explain timed out in a comment
            dag_timed_out = ...
            if dag_folder_different or dag_timed_out:
                ....
```

It's not perfect - you can't short circuit, so you always calculate both conditions - but worth it for readability imo.